content/4_Gold/Combinatorics.mdx

```diff
@@ -0,0 +1,280 @@
+	</Resource>
+	<Resource source="AoPS@Art of Problem Solving" title="Introductory Combinatorics" url = "https://artofproblemsolving.com/wiki/index.php/Combinatorics/Introduction" starred>
+		teaches basic combinatorics concepts
+	</Resource>
+	<Resource source="AoPS@Art of Problem Solving" title="Intermediate Combinatorics" url = "https://artofproblemsolving.com/wiki/index.php/Combinatorics/Introduction">
+		teaches more advanced combinatorics concepts
+	</Resource>
+	<Resource source="CF" title="Expected Value" url = "https://codeforces.com/blog/entry/62690">
+		a good blog about the expected value
+	</Resource>
+	<Resource source="CF" title="Inclusion-Exclusion Principle" url = "https://codeforces.com/blog/entry/64625">
+		a good blog about the inclusion-exclusion principle
+	</Resource>
+</Resources>
+
+If you prefer watching videos instead, here are some options:
+
+<Resources>
+	<Resource source="Youtube" title="Deep Dive into Combinatorics" url = "https://www.youtube.com/playlist?list=PLDcSwjT2BF_W7hSCiSAVk1MmeGLC3xYGg">
+		playlist by mathemaniac
+	</Resource>
+	<Resource source="Youtube" title="MIT 6.042J" url = "https://www.youtube.com/playlist?list=PLDcSwjT2BF_W7hSCiSAVk1MmeGLC3xYGg">
+		lectures 16-23
+	</Resource>
+	<Resource source="Youtube" title="Sums and Expected Value #1" url = "https://www.youtube.com/watch?v=U_h3IjreRek">
+		Errichto video regarding expected value and sums of subsets
+	</Resource>
+</Resources>
+
+## Binomial Coefficients
+
+<FocusProblem problem="sample" />
+
+The **binomial coefficient** $\binom{n}{k}$ (pronounced as "$n$ choose $k$" or sometimes written as ${}_nC_k$) represents the number of ways to choose a subset of $k$ elements from a set of $n$ elements. For example, $\binom{4}{2} = 6$, because the set $\{1,2,3,4\}$ has $6$ subsets of $2$ elements: 
+
+$$
+\{1, 2\}, \{1, 3\}, \{1, 4\}, \{2, 3\}, \{2, 4\}, \{3, 4\}
+$$
+
+There are two ways to calculate binomial coefficients:
+
+### Method 1: Pascal's Triangle (Dynamic Programming) - $\mathcal{O}(n^2)$
+
+Binomial coefficients can be recursively calculated as follows:
+
+$$
+    \binom{n}{k} = \binom{n - 1}{k - 1} + \binom{n - 1}{k}
+$$
+
+The intuition behind this is to fix an element $x$ in the set and choose $k − 1$ elements from $n − 1$ elements if $x$ is included in the set or choose $k$ elements from $n − 1$ elements, otherwise. 
+
+The base cases for the recursion are:
+
+$$
+	\binom{n}{0} = \binom{n}{n}  = 1
+$$
+
+because there is always exactly one way to construct an empty subset and a subset that contains all the elements.
+
+This recursive formula is commonly known as [Pascal's Triangle](https://brilliant.org/wiki/pascals-triangle/). 
+
+Additionally, we can optimize this recursion formula from $\mathcal{O}(2^n)$ to $\mathcal{O}(n^2)$ using [Dynamic Programming](/gold/intro-dp). You can learn more about how DP is used to optimize the binomial coefficient recursive formula [here](https://www.geeksforgeeks.org/binomial-coefficient-dp-9/).
```

Comment:
Additionally, we can optimize this recursion formula from $\mathcal{O}(2^n)$ to $\mathcal{O}(n^2)$ using [dynamic programming](/gold/intro-dp) (DP). You can learn more about how DP is used to optimize the binomial coefficient recursive formula [here](https://www.geeksforgeeks.org/binomial-coefficient-dp-9/).